From d084b626335a540f713c11428d6a5b6cc310f045 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Mon, 21 Sep 2015 15:37:08 +0800
Subject: [PATCH 0457/5242] MLK-11594 video: mxsfb: correct the pm runtime
 logic

commit  5dcff81bd70f223090709673a29a9467a7fc9ff2 from
https://source.codeaurora.org/external/imx/linux-imx.git

The pm power's usage_count should be first added by
one before putting it into suspend state. Otherwise
the usage_count will be negative and the power cannot
be put into suspend state correctly.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
(cherry picked from commit a48460b41399112f5d3ae03b30e3f9e885346ff7)

Conflicts:
	drivers/video/mxsfb.c
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 4114db0..840973e 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -893,9 +893,7 @@ static int mxsfb_blank(int blank, struct fb_info *fb_info)
 	case FB_BLANK_NORMAL:
 		if (host->enabled) {
 			mxsfb_disable_controller(fb_info);
-			pm_runtime_put_noidle(&host->pdev->dev);
 			pm_runtime_put_sync_suspend(&host->pdev->dev);
-			pm_runtime_get_noresume(&host->pdev->dev);
 		}
 
 		clk_disable_disp_axi(host);
@@ -1475,6 +1473,7 @@ static int mxsfb_probe(struct platform_device *pdev)
 		writel(0, host->base + LCDC_CTRL);
 		mxsfb_set_par(fb_info);
 		mxsfb_enable_controller(fb_info);
+		pm_runtime_get_sync(&host->pdev->dev);
 	}
 
 	ret = register_framebuffer(fb_info);
-- 
1.7.9.5

